src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php

```diff
@@ -64,19 +73,18 @@ public function handlePersist(PersistEvent $event)
 
         $propertyName = $this->encoder->systemName(self::FIELD);
 
-        if ($node->hasProperty($propertyName)) {
-            return;
-        }
-
         $parent = $node->getParent();
-        $nodeCount = count($parent->getNodes());
-        $order = ($nodeCount + 1) * 10;
+        $count = 0;
+        foreach ($parent->getNodes() as $childNode) {
```

Comment:
You can get the node by injecting the document registry.

    $this->registry->getNodeForDocument($document); // or similar

It is extremely fast and in anycase this is not a performance sensative
operation.

On Thu, Jul 16, 2015 at 12:32:45AM -0700, Daniel Rotter wrote:
>    In [1]src/Sulu/Component/Content/Document/Subscriber/OrderSubscriber.php:
> 
>  >          $parent = $node->getParent();
>  > -        $nodeCount = count($parent->getNodes());
>  > -        $order = ($nodeCount + 1) * 10;
>  > +        $count = 0;
>  > +        foreach ($parent->getNodes() as $childNode) {
> 
>    [2]@dantleech I am missing the node in the reorder event... Is this
>    something that should be added in the RegistratorSubscriber? I would have
>    handled it the same way as handleNodeFromRegistry, but just with the
>    ReorderEvent (which has to be extended by a node also, so is it a good
>    idea to let it inherit from the AbstractMappingEvent?).
> 
>    —
>    Reply to this email directly or [3]view it on GitHub.
> 
>    Reverse link: [4]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu-io/sulu/pull/1378#discussion_r34762103
>    2. https://github.com/dantleech
>    3. https://github.com/sulu-io/sulu/pull/1378/files#r34762103
>    4. https://github.com/sulu-io/sulu/pull/1378/files#r34762103